collector/database/chart.php

```diff
@@ -0,0 +1,76 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+use tool_cloudmetrics\metric;
+
+/**
+ * Shows a chart of recorded metrics.
+ *
+ * @package   cltr_database
+ * @author    Jason den Dulk <jasondendulk@catalyst-au.net>
+ * @copyright 2022, Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+require_once(__DIR__.'/../../../../../config.php');
+require_once($CFG->libdir.'/adminlib.php');
+
+admin_externalpage_setup('cltr_database_chart');
+
+$context = context_system::instance();
+
+$url = new moodle_url('/admin/tool/cloudmetrics/collector/database/chart.php');
+
+$PAGE->set_context($context);
+$PAGE->set_url($url);
+
+$metricname = optional_param('metric', 'activeusers', PARAM_ALPHANUMEXT);
+
+$metrics = metric\manager::get_metrics();
+$metriclabels = [];
+foreach ($metrics as $metric) {
+    $metriclabels[$metric->get_name()] = $metric->get_label();
+}
+
+$select = new \single_select(
+    $url,
+    'metric',
+    $metriclabels,
+    $metricname
+);
+$select->set_label(get_string('select_metric_for_display', 'cltr_database'));
+
+$records = $DB->get_records('cltr_database_metrics', ['name' => $metricname], 'time asc', 'id, time, value');
+
+$values = [];
+$labels = [];
+
+foreach ($records as $record) {
+    $values[] = (float) $record->value;
+    $labels[] = userdate($record->time, get_string('strftimedatetimeshort'), $CFG->timezone);
+}
+
+$chartseries = new \core\chart_series($metriclabels[$metricname], $values);
+
+$chart = new \core\chart_line();
+$chart->add_series($chartseries);
+$chart->set_labels($labels);
+
+$output = $PAGE->get_renderer('core');
```

Comment:
$OUTPUT is the core renderer by default, so you can just use the $OUTPUT global for all of these

Reply:
done